ui/src/components/imquery/AddPropertyDialog.vue

```diff
@@ -63,29 +82,60 @@ watch(
   }
 );
 
+watch(
+  () => cloneDeep(selectedProperty.value),
+  newValue => {
+    if (isObjectHasKeys(selectedProperty.value)) {
+      whereOrMatch.value = buildProperty(selectedProperty.value as any);
+      if (isObjectHasKeys(whereOrMatch.value, ["typeOf", "where"])) {
+        editWhere.value = getEditWhere(whereOrMatch.value.where![0]!);
+        editWhereDMIri.value = getEditWhereDMIri(whereOrMatch.value.where![0]!) ?? (whereOrMatch.value as Match).typeOf?.["@id"];
+      } else {
+        editWhere.value = getEditWhere(whereOrMatch.value);
+        editWhereDMIri.value = getEditWhereDMIri(whereOrMatch.value);
+      }
+    }
+  }
+);
+
 onMounted(() => {
   if (isObjectHasKeys(props.match, ["property"]) && isArrayHasLength(props.match!.where)) editMatch.value.where = cloneDeep(props.match!.where);
 });
 
-function onSelectedUpdate(selected: TreeNode[]) {
-  selectedProperties.value = selected;
+async function save() {
+  if (isObjectHasKeys(whereOrMatch.value, ["typeOf", "where"])) {
+    emit("onMatchAdd", whereOrMatch.value as Match);
+  } else emit("onPropertyAdd", whereOrMatch.value as Where);
+  visible.value = false;
 }
 
-async function save() {
-  editMatch.value.where = [];
-  const properties = buildMatchesFromProperties(selectedProperties.value as any);
-  if (isArrayHasLength(properties)) {
-    const parts = properties[0].description?.split("-");
-    if (parts && parts.length === 5) {
-      for (const match of properties as Match[]) {
-        match.description = getNameFromRef(match.typeOf);
-      }
-      emit("onMatchAdd", properties as Match[]);
-    } else {
-      emit("onPropertyAdd", properties);
+function getEditWhere(whereMatch: any) {
+  const found: any[] = [];
+  getEditWhereRecursively(whereMatch, found);
+  if (isArrayHasLength(found)) return found[0];
+}
+
+function getEditWhereRecursively(where: Where, found: any[]) {
+  if (where.match?.where) {
+    for (const nestedWhere of where.match?.where) {
+      getEditWhereRecursively(nestedWhere, found);
+    }
+  } else found.push(where);
+}
+
+function getEditWhereDMIri(whereMatch: any) {
+  const found: string[] = [];
+  getEditWhereDMIriRecursively(whereMatch, found);
+  if (isArrayHasLength(found)) return found[0];
+}
+
+function getEditWhereDMIriRecursively(where: Where, found: any[]) {
+  if (where.match?.where) {
+    found[0] = where.match.typeOf?.["@id"];
```

Comment:
why not pushing to array? Looks like it will recursively overwrite.

Reply:
that's intended. there is only one datamodel iri that is needed which is in the deepest match